File: jruby-project/collections.rb
# Arrays:

a = [1,2,3,4]
a << 5
a.include? 3 #true

#Sets:

require 'set'
s = [1,2,3,4].to_set
s2 = Set.new [1,2,3,4]
s2 << 5 #Add an item
s2.include? 4 #true

#Hashes:
# :One, :Two, and :Three are Ruby Symbols
# much like Scala Symbols and Clojure Keywords

h = {1 => :One, 2 => :Two, 3 => :Three}
h[4] = :Four #Add an item
puts(h[1])
h.include? 1 #true
h.has_key? 3 #true

File: clojure-project/src/collections.clj
;; Lists (not typical):

(def my-list '(1 2 3 4 5))
(conj my-list 6) ;; placed in the beginning


;;Vectors (Preferred over lists):

(def my-vector [1 2 3 4])
(def my-vector-2 (vector 1 2 3 4))
(conj my-vector 5) ;; placed at the end
(some #{4} my-vector) ;; does my-vector contain item, a bit weird
                      ;; that's because strong possibility you need 
                      ;; a set!
;;Sets:
(def my-set (set [1 2 3 4]))
(def my-set-2 #{1 2 3 4})
(conj my-set 5)
(contains? my-set 4)

;;Maps:

;; :One, :Two, and :Three are Clojure Keywords
;; much like Ruby Symbols and Scala Symbols

(def my-map {1 :One 2 :Two 3 :Three}); :before are keywords
(assoc my-map 4 :Four) ;Add an item to the map
(contains? my-map 3)
(my-map 2) ;Retreive by key
